proto/proto-backplane-grpc/src/main/proto/deephaven/proto/object.proto

```diff
@@ -12,7 +12,67 @@ option go_package = "github.com/deephaven/deephaven-core/go/internal/proto/objec
+     * This API lets a client open a stream to a particular object on the server, to be mediated by
+     * a server side plugin. In theory this could effectively be used to "tunnel" a custom gRPC call,
+     * but in practice there are a few deliberate shortcomings that still make this possible, but not
+     * trivial.
+     *
+     * The first message sent to the server is expected to have a ConnectRequest, indicating which
+     * export ticket to connect to. It is an error for the server to not have a plugin defined for that
+     * object type.
+     *
+     * Subsequent requests to the server, and all responses to the client, will be a payload of bytes
+     * and an arbitrary number of tickets, referring to server side objects that one side wishes to
+     * reference in its payload.
+     *
+     * Presently it is required that the server respond immediately, at least to acknowledge that the
+     * object was correctly contacted (as opposed to waiting for a pending ticket, or dealing with
+     * network lag, etc). This is a small (and possibly not required, but convenient) departure from
+     * a offering a gRPC stream (a server-streaming or bidi-streaming call need not send a message
+     * right away).
+     *
+     * Presently there is no explicit "close" message to send, but plugin implementations can devise
+     * their own "half-close" protocol if they so choose. For now, if one end closes the connection,
+     * the other is expected to follow suit by closing their end too.
+     *
+     * (address stream failures?)
+     *
+     * Finally, addressing a broader point of unsuitability for general gRPC-over-gRPC tunneling:
+     * there is a lot more to gRPC than just specifying a path and sending payloads to it, such as
+     * how to specify metadata. In theory, we could handle the metadata as another field (only sent
+     * in the first/last payloads), or as a separate plugin-defined type of payload (analogous to the
+     * grpc-websockets implementations), but we still run into issues where this user-defined gRPC
+     * service needs to coexist with the rest of the Deephaven platform, supporting at least authentication,
+     * and for browsers, our custom streaming implementation. The nested gRPC implementation would
+     * need to be at some level aware that it is already stateful to take advantage of this, rather
+     * than somehow redefine it.
+     */
+    rpc MessageStream(stream StreamRequest) returns (stream StreamResponse) {}
+
+    /*
+     * Half of the browser-based (browser's can't do bidirectional streams without websockets)
+     * implementation for MessageStream.
+     */
+    rpc OpenMessageStream(StreamRequest) returns (stream StreamResponse) {}
+    /*
+     * Other half of the browser-based implementation for MessageStream.
+     */
+    rpc NextMessageStream(StreamRequest) returns (BrowserNextResponse) {}
```

Comment:
Do we need some id so that the NextMessageStream requests can be tied to an OpenMessageStream?

Reply:
No, this is handled as metadata (aka http headers), see `io.deephaven.server.browserstreaming.BrowserStreamInterceptor` and `io.deephaven.web.client.api.barrage.stream.BiDiStream.EmulatedBiDiStream#makeHeaders` for the wrappers on either side of the wire that add these transparently, transforming any Open/Next grpc calls into proper bidi stream calls.